A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+verification", but in a mesh scenario may be known as "server authorization" as
+the server's certificate does not include the hostname but other identity
+information.
+
+While servers authenticated clients (i.e., the client's identity was verified),
+A29 did not include mechanism to limit which clients were authorized to access a
+server and for which gRPC methods (i.e., whether the client was permitted).
+
+Envoy defines the [RBAC HTTP filter][RBAC filter] that is well-suited for
+limiting which nodes can access which services within a mesh as encouraged by
+the principal of least privilege.
+
+Resources in REST are present in the HTTP path so it would seem RBAC could be
+used in non-gRPC environments as a precise service-level resource authz engine.
+However, this 1) requires another filter to perform end-user authentication (as
+TLS just provides service-level authn) and 2) does not scale as RBAC provides no
+mechanism for loading rules on-demand. Since gRPC encodes service-level
+resource identifiers in the request message, RBAC does not have access to the
+requested resource for gRPC traffic.
+
+### Related Proposals:
+
+* [A36: xDS-Enabled Servers][A36]
+* [A29: xDS-Based Security for gRPC Clients and Servers][A29]
+* [A39: xDS HTTP Filter Support][A39]
+
+[A36]: https://github.com/grpc/proposal/blob/master/A36-xds-for-servers.md
+<!-- FIXME: replace with final link once merged -->
+[A29]: https://github.com/grpc/proposal/pull/184
+[A39]: https://github.com/grpc/proposal/blob/master/A39-xds-http-filters.md
+
+## Proposal
+
+Support the [xDS RBAC HTTP filter][RBAC filter] as a registered filter type on
+xDS-enabled gRPC Servers, and support its `RBACPerRoute` configuration override.
+This does _not_ include [RBAC network filter][] support nor running the filter
+on gRPC clients. xDS v2 support is not necessary.
+
+Supporting the RBAC HTTP filter on server-side leverages [A36: xDS-Enabled
+Servers][A36] and [A39: xDS HTTP Filter Support][A39]. Server-side HTTP filters
+are less common than client-side at this point in time, so this may be the first
+usage of `RouteConfiguration` on server-side and so likely involves adding more
+complete support for A39 by creating+executing filters on server-side and
+processing `VirtualHost`s and `RoutMatch`es to determine which configuration
+should be provided to each filter. A39 should be consulted for the expected
+behavior.
+
+The core policy matching logic should be split into an "RBAC engine" to allow
+reuse with non-xDS environments. However, there is no requirement that the RBAC
+engine have a specialized API; it could simply be an interceptor.
+
+The [RBAC policy][] has many fields. All current (Envoy-implemented) fields will
+be considered in gRPC. However, some fields may have pre-determined values or
+behavior. At this time, if the `RBAC.action` is `Action.LOG` then the policy
+will be completely ignored, as if RBAC was not configurated. CEL fields are not
```

Comment:
For `Action.LOG` the doc says "All requests are allowed." which is what Envoy does presumably. If gRPC is going to completely ignore the policy then it's possible request(s) will be denied (due to the default or another policy) and we'll have difference in behavior between Envoy and gRPC for the same set of control plane policies. That's undesirable, isn't it?